ln-simln-jamming/src/attacks/utils.rs

```diff
@@ -176,9 +172,371 @@ pub async fn build_reputation<C: Clock + InstantClock, R: ReputationMonitor>(
         .ok_or(format!("channel {} not found", last_hop_channel))?
         .outgoing_reputation;
 
-    if attacker_reputation as u64 > (target_revenue as u64 + htlc_amounts + htlc_risk) {
+    if attacker_reputation as u64 > (target_revenue as u64 + htlc_risk) {
         Ok(total_fees_paid)
     } else {
         Err("could not build reputation".into())
     }
 }
+
+// Calculates the fee amount that will need to be paid to build sufficient reputation.
+// Allows for small buffer to account for decayed average when a reputation check is done
+// afterwards.
+fn fee_to_build_reputation(
+    htlc_routes: &[Route],
+    forward_params: &ForwardManagerParams,
+    target_hop_pubkey: PublicKey,
+    channel_revenue: i64,
+    peer_reputation: i64,
+    fee_buffer: u64,
+) -> (u64, u64) {
+    let mut total_htlc_risk = 0;
+    for route in htlc_routes {
+        for path in &route.paths {
+            if let Some(target_idx) = path
+                .hops
+                .iter()
+                .position(|hop| hop.pubkey == target_hop_pubkey)
+            {
+                // Add up the cltv_delta(s) from the target hop to get the total number of blocks
+                // until the htlc can be resolved. Here we assume the current height is 0.
+                //
+                // We use `build_route_from_hops` from LDK to build the routes for the
+                // payments. However, that method adds a random CLTV offset so the route we got
+                // here could have a different `cltv_expiry_delta` from a route we got another time
+                // after we have built reputation. Differences in the cltv deltas in some cases
+                // could case the calculated fee here to not be enough. Hence, we default to
+                // a reasonably higher value of 200 which overpays a little but should be enough
+                // to cover the random offsets.
+                let cltv_expiry: u32 = path.hops[target_idx..].iter().len() as u32 * 200;
+
+                let target_hop = &path.hops[target_idx];
+                let current_htlc_risk =
+                    forward_params.htlc_opportunity_cost(target_hop.fee_msat, cltv_expiry);
+
+                total_htlc_risk += current_htlc_risk;
+            }
+        }
+    }
+
+    let threshold = if channel_revenue - peer_reputation > 0 {
+        channel_revenue - peer_reputation
+    } else {
+        0
+    } as u64;
```

Comment:
Shouldn't this include the `total_htlc_risk`

Eg:
* `channel_revenue` = 100
* `peer_reputation` = 150
* `total_htlc_risk` = 40

`threshold` = 0 (since 100 -150 < 0), but then we'll go on to pay 40 for `total_htlc_risk` even though we actually have enough reputation for it?

Also _could_ technically overflow to use `saturating_sub`?